ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -612,22 +612,76 @@ - (NSArray*)serializePacksArray:(NSArray<MGLOfflinePack*>*)packs
     [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
         RCTMapboxGL *mapView = viewRegistry[reactTag];
         if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
-            NSNumber *screenCoordX = options[@"screenCoordX"];
-            NSNumber *screenCoordY = options[@"screenCoordY"];
-            if (!screenCoordX || !screenCoordY) {
+            NSDictionary *pointDict = options[@"point"];
+            NSDictionary *rect = options[@"rect"];
+            if (!pointDict) {
                 // TODO: figure out best way to error this. Should we be using RCTPromise rather than a callback?
 //                reject(@"invalid_arguments", @"queryRenderedFeatures(): screenCoordX and screenCoordY are required.", nil);
                 return;
             }
+            NSNumber *screenCoordX = pointDict[@"screenCoordX"];
+            NSNumber *screenCoordY = pointDict[@"screenCoordY"];
+            NSSet<NSString *> *styleLayerIdentifiers = options[@"layers"];
+
+            CLLocationCoordinate2D location = [_map convertPoint:[sender locationInView:_map] toCoordinateFromView:_map];
+            CGPoint screenCoord = [sender locationInView:_map];
+
+
             CGPoint point = CGPointMake(screenCoordX.floatValue, screenCoordY.floatValue);
 
-            NSArray * features = [mapView visibleFeaturesAtPoint:point];
+            NSArray *features;
+            if (!styleLayerIdentifiers) {
+                features = [mapView visibleFeaturesAtPoint:point];
+            } else {
+                features = [mapView visibleFeaturesAtPoint:point inStyleLayersWithIdentifiers:styleLayerIdentifiers];
```

Comment:
You can call this method for both cases. Pass in `nil` as the second parameter to get the same behavior as `-visibleFeaturesAtPoint:`.